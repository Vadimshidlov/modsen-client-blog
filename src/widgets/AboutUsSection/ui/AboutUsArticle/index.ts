export { AboutUsArticle } from '@/widgets/AboutUsSection/ui/AboutUsArticle/AboutUsArticle';
