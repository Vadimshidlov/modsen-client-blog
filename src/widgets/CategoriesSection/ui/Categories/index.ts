export { Categories } from '@/widgets/CategoriesSection/ui/Categories/Categories';
