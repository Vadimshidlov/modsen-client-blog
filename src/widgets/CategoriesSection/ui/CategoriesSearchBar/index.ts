export { CategoriesSearchBar } from '@/widgets/CategoriesSection/ui/CategoriesSearchBar/CategoriesSearchBar';
