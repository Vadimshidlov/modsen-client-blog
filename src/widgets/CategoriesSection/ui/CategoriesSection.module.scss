.categoriesSection {
  display: flex;
  margin-bottom: 128px;
  padding-top: 102px;
  // column-gap: 147px;
  column-gap: 40px;
  justify-content: space-between;

  @media (max-width: 1250px) {
    flex-direction: column-reverse;
  }

  @media (max-width: $tablet) {
    padding-top: 51px;
    margin-bottom: 64px;
  }

  @media (max-width: $mobile) {
    padding-top: 25px;
    margin-bottom: 32px;
  }

  .postsContainer {
    display: flex;
    flex-direction: column;
    row-gap: 32px;
    max-width: 844px;

    .postsNotFountTitle {
      @include headingThree;

      @media (max-width: $tablet) {
        @include headingThree;
      }

      @media (max-width: $mobile) {
        @include headingFive;
      }
    }
  }

  .searchBarWrapper {
    max-width: 300px;
    display: flex;
    flex-direction: column;

    @media (max-width: 1250px) {
      max-width: 100%;
    }
  }
}
