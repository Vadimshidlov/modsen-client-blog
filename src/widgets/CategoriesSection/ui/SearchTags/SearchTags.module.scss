.searchTagsContainer {
  .searchTagsTitle {
    margin-bottom: 24px;
    @include headingTwo;

    @media (max-width: $tablet) {
      margin-bottom: 20px;
      @include headingFour;
    }
  }

  .searchTagsList {
    display: flex;
    row-gap: 24px;
    flex-wrap: wrap;
    justify-content: space-between;
    margin-bottom: 20px;

    @media (max-width: $tablet) {
      row-gap: 10px;
    }

    .searchTagItem {
      font-size: $fz14;
      font-weight: $bold;
      line-height: 32px;
      padding: 8px 38px;
      border: 2px solid $grey;
      border-radius: 28px;
      cursor: pointer;
      transition: all .5s ease;

      &:hover {
        background: $hoverYellow;
      }

      &.activeTag:hover {
        background: $yellow;
      }

      @media (max-width: $tablet) {
        padding: 4px 25px;
      }
    }
  }
}

.activeTag {
  background: $hoverYellow;
}
