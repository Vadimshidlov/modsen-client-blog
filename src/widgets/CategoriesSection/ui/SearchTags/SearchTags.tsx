import React from 'react';
import styles from '@/widgets/CategoriesSection/ui/SearchTags/SearchTags.module.scss';
import { SEARCH_TAGS } from '@/widgets/CategoriesSection/constants/constants';
import { useTranslations } from 'next-intl';

export type SearchTagsPropsTypes = {
  selectTags: string[];
  setSelectTags: React.Dispatch<React.SetStateAction<string[]>>;
};

export function SearchTags({ selectTags, setSelectTags }: SearchTagsPropsTypes) {
  const t = useTranslations('SearchTags');

  return (
    <div className={styles.searchTagsContainer}>
      <h2 className={styles.searchTagsTitle}>{t('title')}</h2>
      <ul className={styles.searchTagsList}>
        {SEARCH_TAGS.map((searchTag) => {
          const isActiveTag = selectTags.includes(searchTag);

          return (
            <li
              key={searchTag}
              className={`${styles.searchTagItem} ${isActiveTag ? `${styles.activeTag}` : ''}`}
              onClick={() => {
                const preparedTagsList = selectTags.includes(searchTag)
                  ? selectTags.filter((tag) => tag !== searchTag)
                  : [...selectTags, searchTag];

                setSelectTags([...preparedTagsList]);
              }}
            >
              {t(`${searchTag}`)}
            </li>
          );
        })}
      </ul>
    </div>
  );
}
