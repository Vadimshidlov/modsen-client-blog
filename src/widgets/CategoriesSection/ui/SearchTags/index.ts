export { SearchTags } from '@/widgets/CategoriesSection/ui/SearchTags/SearchTags';
