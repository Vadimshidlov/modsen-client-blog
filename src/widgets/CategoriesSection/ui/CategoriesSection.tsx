'use client';

import { useState } from 'react';
import styles from '@/widgets/CategoriesSection/ui/CategoriesSection.module.scss';
import container from '@/shared/styles/container.module.scss';
import { CategoriesSearchBar } from '@/widgets/CategoriesSection/ui/CategoriesSearchBar';
import { POSTS } from '@/shared/constants/posts';
import { PostItem } from '@/widgets/PostItem';
import { Categories } from '@/widgets/CategoriesSection/ui/Categories';
import { SearchTags } from '@/widgets/CategoriesSection/ui/SearchTags';
import { useTranslations } from 'next-intl';
import { CategoriesSectionPropsType } from '@/shared/types';

export function CategoriesSection({ selectedCategory }: CategoriesSectionPropsType) {
  const t = useTranslations('CategoriesSection');

  const [selectTags, setSelectTags] = useState<string[]>([]);

  const filteredPosts = POSTS.filter(
    (postData) =>
      postData.category.toLowerCase() === selectedCategory.toLowerCase() &&
      (selectTags.length === 0 || postData.tags.some((tag) => selectTags.includes(tag))),
  );

  return (
    <section className={`${styles.categoriesSection} ${container.container}`}>
      <div className={styles.postsContainer}>
        {filteredPosts.length === 0 && (
          <span className={styles.postsNotFountTitle}>{t('notFoundMessage')}</span>
        )}
        {filteredPosts.length !== 0 &&
          filteredPosts
            .slice(0, 3)
            .map(({ id, img, category, title, text }) => (
              <PostItem category={category} id={id} img={img} text={text} title={title} key={id} />
            ))}
      </div>
      <div className={styles.searchBarWrapper}>
        <SearchTags selectTags={selectTags} setSelectTags={setSelectTags} />
        <CategoriesSearchBar />
        <Categories />
      </div>
    </section>
  );
}
