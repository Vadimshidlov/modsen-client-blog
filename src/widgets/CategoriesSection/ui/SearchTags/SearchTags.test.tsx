import React from 'react';
import { render } from '@testing-library/react';
import { SearchTags } from '@/widgets/CategoriesSection/ui/SearchTags';

jest.mock('next-intl', () => ({
  useTranslations: jest.fn().mockReturnValue((key: string) => `Mocked ${key}`),
}));

describe('SearchTags tests', () => {
  test('should matches snapshot', () => {
    const selectTags: string[] = [];
    const setSelectTags = jest.fn();

    const { container } = render(
      <SearchTags selectTags={selectTags} setSelectTags={setSelectTags} />,
    );
    expect(container).toMatchSnapshot();
  });
});
