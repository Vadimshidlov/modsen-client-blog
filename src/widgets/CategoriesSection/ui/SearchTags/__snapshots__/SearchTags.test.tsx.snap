// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchTags tests should matches snapshot 1`] = `
<div>
  <div
    class="searchTagsContainer"
  >
    <h2
      class="searchTagsTitle"
    >
      Mocked title
    </h2>
    <ul
      class="searchTagsList"
    >
      <li
        class="searchTagItem "
      >
        Mocked Business
      </li>
      <li
        class="searchTagItem "
      >
        Mocked Experience
      </li>
      <li
        class="searchTagItem "
      >
        Mocked Screen
      </li>
      <li
        class="searchTagItem "
      >
        Mocked Technology
      </li>
      <li
        class="searchTagItem "
      >
        Mocked Marketing
      </li>
      <li
        class="searchTagItem "
      >
        Mocked Life
      </li>
    </ul>
  </div>
</div>
`;
